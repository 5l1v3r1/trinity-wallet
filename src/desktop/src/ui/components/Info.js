import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'ui/components/Icon';
import css from './info.scss';

/**
 * Info block component
 */
export default class Info extends React.PureComponent {
    static propTypes = {
        /** Info block content */
        children: PropTypes.node.isRequired,
        /** Determines whether to display icon */
        displayIcon: PropTypes.bool,
    };

    static defaultProps = {
        displayIcon: true,
    };

    render() {
        const { children, displayIcon } = this.props;

        return (
            <div className={css.wrapper}>
                {displayIcon && <Icon icon="info" size={28} />}
                {children}
            </div>
        );
    }
}
