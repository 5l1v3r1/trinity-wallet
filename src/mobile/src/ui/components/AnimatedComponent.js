import last from 'lodash/last';
import merge from 'lodash/merge';
import React, { Component } from 'react';
import { Animated, Easing } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { width } from 'libs/dimensions';

class AnimatedComponent extends Component {
    static propTypes = {
        /** Component to animate */
        children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
        /** Determines whether to animate in on mount */
        animateOnMount: PropTypes.bool,
        /** Delay for animation trigger */
        delay: PropTypes.number,
        /** Style of animation in */
        animationInType: PropTypes.array,
        /** Style of animation out */
        animationOutType: PropTypes.array,
        /** Duration of animation */
        duration: PropTypes.number,
        /** @ignore */
        navStack: PropTypes.array.isRequired,
        /** Style prop */
        style: PropTypes.oneOfType([PropTypes.object, PropTypes.array, PropTypes.number]),
    };

    static defaultProps = {
        animateOnMount: true,
        delay: 0,
        animationInType: ['fadeIn'],
        animationOutType: ['fadeOut'],
        duration: 300,
        triggerAnimationIn: false,
        triggerAnimationOut: false,
    };

    constructor(props) {
        super(props);
        this.fadeValue = new Animated.Value(1);
        this.slideValue = new Animated.Value(0);
        this.animatedStyle = {};
        // Slide animations are reversed when popping from navigation stack
        this.reverseSlideIn = false;
        this.reverseSlideOut = false;
        this.iniatialiseAnimations(props.animationInType);
        this.screen = last(props.navStack);
    }

    componentDidMount() {
        // Animate in on mount
        if (this.props.animateOnMount) {
            this.animateIn(this.props.delay);
        }
    }

    componentWillReceiveProps(newProps) {
        // Animate out if pushing from current screen
        if (this.props.navStack.length < newProps.navStack.length && this.screen === last(this.props.navStack)) {
            this.reverseSlideOut = false;
            this.iniatialiseAnimations(this.props.animationOutType);
            this.animateOut();
        }
        // Animate out if popping from current screen
        if (
            this.props.navStack.length > newProps.navStack.length &&
            this.screen === last(this.props.navStack) &&
            newProps.navStack.length !== 1
        ) {
            this.reverseSlideOut = true;
            this.iniatialiseAnimations(this.props.animationOutType);
            this.animateOut();
        }

        // Animate out if resetting navigator stack
        if (
            this.screen === last(this.props.navStack) &&
            newProps.navStack.length === 1 &&
            last(this.props.navStack) !== last(newProps.navStack)
        ) {
            this.reverseSlideOut = false;
            this.iniatialiseAnimations(this.props.animationOutType);
            this.animateOut();
        }

        // Animate in if popping to current screen
        if (this.props.navStack.length > newProps.navStack.length && this.screen === last(newProps.navStack)) {
            this.reverseSlideIn = true;
            this.iniatialiseAnimations(this.props.animationInType);
            // Allow time for animate out (300ms)
            this.animateIn(this.props.delay + 300);
        }
    }

    /**
     * Gets animated style object for single or multiple animations
     * @method getAnimatedStyle
     * @param {string}
     * @returns {object}
     */
    getAnimatedStyle(animationType) {
        let animatedStyle = {};
        animationType.map((type) => {
            switch (type) {
                case 'fadeIn':
                    animatedStyle = merge({}, animatedStyle, { opacity: this.fadeValue });
                    break;
                case 'fadeOut':
                    animatedStyle = merge({}, animatedStyle, { opacity: this.fadeValue });
                    break;
                case 'slideInRight':
                    animatedStyle = merge({}, animatedStyle, { width, transform: [{ translateX: this.slideValue }] });
                    break;
                case 'slideOutLeft':
                    animatedStyle = merge({}, animatedStyle, { width, transform: [{ translateX: this.slideValue }] });
                    break;
            }
        });
        return animatedStyle;
    }

    /**
     * Gets starting animated value
     * @method getStartingAnimatedValue
     * @param {string}
     * @returns {number}
     */
    getStartingAnimatedValue(animationType) {
        switch (animationType) {
            case 'fadeIn':
                return 0;
            case 'fadeOut':
                return 1;
            case 'slideInRight':
                return this.reverseSlideIn ? -width : width;
            case 'slideOutLeft':
                return 0;
        }
    }

    /**
     * Gets final animated value
     * @method getFinalAnimatedValue
     * @param {string}
     * @returns {number}
     */
    getFinalAnimatedValue(animationType) {
        switch (animationType) {
            case 'fadeIn':
                return 1;
            case 'fadeOut':
                return 0;
            case 'slideInRight':
                return 0;
            case 'slideOutLeft':
                return this.reverseSlideOut ? width : -width;
        }
    }

    /**
     * Gets easing function
     * @method getEasing
     * @param {string}
     * @returns {function}
     */
    getEasing(animationType) {
        switch (animationType) {
            case 'fadeIn':
                return Easing.ease;
            case 'fadeOut':
                return Easing.ease;
            case 'slideOutLeft':
                return Easing.bezier(0.25, 1, 0.25, 1);
            case 'slideInRight':
                return Easing.bezier(0.25, 1, 0.25, 1);
        }
    }

    /**
     * Gets variable pointer for each animation type
     * @method getAnimation
     * @param {string}
     * @returns {number}
     */
    getAnimationPointer(animationType) {
        switch (animationType) {
            case 'fadeIn':
                return this.fadeValue;
            case 'fadeOut':
                return this.fadeValue;
            case 'slideInRight':
                return this.slideValue;
            case 'slideOutLeft':
                return this.slideValue;
        }
    }

    /**
     * Creates animation and gets relevant values
     * @method getAnimation
     * @param {string}
     * @param {number}
     * @returns {array}
     */
    getAnimation(animationType, delay = 0) {
        const animations = [];
        animationType.map((type) => {
            animations.push(
                Animated.timing(this.getAnimationPointer(type), {
                    toValue: this.getFinalAnimatedValue(type),
                    duration: type === 'fadeOut' ? 100 : this.props.duration,
                    delay,
                    easing: this.getEasing(type),
                }),
            );
        });
        return animations;
    }

    /**
     * Assigns animated values to pointers and initialises animation style
     * @method iniatialiseAnimations
     * @param {string}
     */
    iniatialiseAnimations(animationType) {
        animationType.map((type) => {
            switch (type) {
                case 'fadeIn':
                    this.fadeValue = new Animated.Value(this.getStartingAnimatedValue(type));
                    break;
                case 'fadeOut':
                    this.fadeValue = new Animated.Value(this.getStartingAnimatedValue(type));
                    break;
                case 'slideInRight':
                    this.slideValue = new Animated.Value(this.getStartingAnimatedValue(type));
                    break;
                case 'slideOutLeft':
                    this.slideValue = new Animated.Value(this.getStartingAnimatedValue(type));
                    break;
            }
        });
        this.animatedStyle = this.getAnimatedStyle(animationType);
    }

    /**
     * Animate in on page transition
     * @method animateIn
     * @param {number}
     */
    animateIn(delay = this.props.delay) {
        Animated.parallel(this.getAnimation(this.props.animationInType, delay)).start();
    }

    /**
     * Animate out on page transition
     * @method animateOut
     */
    animateOut() {
        Animated.parallel(this.getAnimation(this.props.animationOutType)).start();
    }

    render() {
        const { children, style } = this.props;
        return (
            <Animated.View style={[this.animatedStyle, style ? style : { alignItems: 'center' }]}>
                {children}
            </Animated.View>
        );
    }
}

const mapStateToProps = (state) => ({
    navStack: state.wallet.navStack,
});

export default connect(mapStateToProps)(AnimatedComponent);
