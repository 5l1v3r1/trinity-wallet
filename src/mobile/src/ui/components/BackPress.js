import last from 'lodash/last';
import React, { Component } from 'react';
import { BackHandler, ToastAndroid } from 'react-native';
import { Navigation } from 'react-native-navigation';
import { navigator } from 'libs/navigation';
import RNExitApp from 'react-native-exit-app';
import i18next from 'shared-modules/libs/i18next.js';
import { setSetting } from 'shared-modules/actions/wallet';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { isAndroid } from 'libs/device';

const mapDispatchToProps = {
    setSetting,
};

const mapStateToProps = (state) => ({
    currentSetting: state.wallet.currentSetting,
    currentRoute: last(state.wallet.navStack),
});

/**
 * Handles back button events on Android by wrapping a component
 * @param {Component} C Component to be wrapped
 * @return {Component} A wrapped component
 */
export default function withBackPress(C) {
    class EnhancedComponent extends Component {
        static propTypes = {
            /** @ignore */
            setSetting: PropTypes.func.isRequired,
            /** @ignore */
            currentSetting: PropTypes.string.isRequired,
            /** @ignore */
            currentRoute: PropTypes.string.isRequired,
            /** Component ID */
            componentId: PropTypes.string.isRequired,
        };
        constructor(props) {
            super(props);
            if (isAndroid) {
                Navigation.events().bindComponent(this);
            }
        }

        /**
         * Remove back handler
         *
         * @method componentDidDisappear
         */
        componentDidDisappear() {
            if (this.backHandler) {
                this.backHandler.remove();
            }
        }

        /**
         * Add back handler
         *
         * @method componentDidAppear
         */
        componentDidAppear() {
            this.backHandler = BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
        }

        /**
         * On back press, pop current route
         *
         * @method withBackPressPopRoute
         */
        withBackPressPopRoute() {
            navigator.pop(this.props.componentId);
        }

        /**
         * On back press, display alert. On second back press, close app
         *
         * @method withBackPressCloseApp
         */
        withBackPressCloseApp() {
            if (this.lastBackPressed && this.lastBackPressed + 2000 >= Date.now()) {
                RNExitApp.exitApp();
            }
            this.lastBackPressed = Date.now();
            ToastAndroid.show(i18next.t('global:pressBackAgain'), ToastAndroid.SHORT);
        }

        /**
         * On back press, navigate to appropriate setting menu
         *
         * @method withBackPressNavigateSettings
         */
        withBackPressNavigateSettings(currentSetting) {
            switch (currentSetting) {
                case 'mainSettings':
                    this.withBackPressCloseApp();
                    return true;
                case 'modeSelection':
                case 'themeCustomisation':
                case 'currencySelection':
                case 'languageSelection':
                case 'accountManagement':
                case 'securitySettings':
                case 'advancedSettings':
                case 'about':
                    this.props.setSetting('mainSettings');
                    return true;
                case 'nodeSelection':
                case 'addCustomNode':
                case 'manualSync':
                case 'snapshotTransition':
                case 'pow':
                case 'autoPromotion':
                    this.props.setSetting('advancedSettings');
                    return true;
                case 'viewSeed':
                case 'viewAddresses':
                case 'editAccountName':
                case 'deleteAccount':
                case 'addNewAccount':
                    this.props.setSetting('accountManagement');
                    return true;
                case 'addExistingSeed':
                    this.props.setSetting('addNewAccount');
                    return true;
                case 'changePassword':
                    this.props.setSetting('securitySettings');
                    return true;
                default:
                    break;
            }
            return true;
        }

        /**
         * Choose appropriate action on back press
         *
         * @method handleBackPress
         */
        handleBackPress = () => {
            const { currentSetting, currentRoute } = this.props;
            switch (currentRoute) {
                case 'home':
                    return this.withBackPressNavigateSettings(currentSetting);
                case 'languageSetup':
                case 'onboardingComplete':
                case 'login':
                    return this.withBackPressCloseApp();
                default:
                    return this.withBackPressPopRoute();
            }
        };

        render() {
            return <C {...this.props} />;
        }
    }

    return connect(mapStateToProps, mapDispatchToProps)(EnhancedComponent);
}
