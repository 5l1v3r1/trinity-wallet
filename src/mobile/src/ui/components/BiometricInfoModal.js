import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import { Styling } from 'ui/theme/general';
import { width, height } from 'libs/dimensions';
import { leaveNavigationBreadcrumb } from 'libs/bugsnag';
import SingleFooterButton from './SingleFooterButton';

const styles = StyleSheet.create({
    modalContainer: {
        alignItems: 'center',
        width,
        height,
        justifyContent: 'flex-end',
    },
    modalContent: {
        justifyContent: 'space-between',
        alignItems: 'center',
        width,
        height: height - Styling.topbarHeight,
    },
    textContainer: {
        width: width - width / 10,
        paddingBottom: height / 22,
    },
    infoText: {
        fontFamily: 'SourceSansPro-Light',
        fontSize: Styling.fontSize3,
        backgroundColor: 'transparent',
        textAlign: 'left',
    },
    infoTextBold: {
        fontFamily: 'SourceSansPro-Bold',
        fontSize: Styling.fontSize3,
        backgroundColor: 'transparent',
    },
});

export default class BiometricInfoModal extends PureComponent {
    static propTypes = {
        /** @ignore */
        theme: PropTypes.object.isRequired,
        /** @ignore */
        t: PropTypes.func.isRequired,
        /** Hides modal */
        hideModal: PropTypes.func.isRequired,
    };

    componentDidMount() {
        leaveNavigationBreadcrumb('BiometricInfoModal');
    }

    render() {
        const { theme: { body }, t } = this.props;
        return (
            <View style={styles.modalContainer}>
                <View style={[styles.modalContent, { backgroundColor: body.bg }]}>
                    <View style={{ flex: 1 }} />
                    <View style={styles.textContainer}>
                        <Text style={[styles.infoTextBold, { color: body.color }, { paddingTop: height / 40 }]}>
                            {t('login:whyBiometricDisabled')}
                        </Text>
                        <Text style={[styles.infoText, { color: body.color }, { paddingTop: height / 60 }]}>
                            {t('login:whyBiometricDisabledExplanation')}
                        </Text>
                    </View>
                    <View style={{ flex: 1 }} />
                    <SingleFooterButton onButtonPress={() => this.props.hideModal()} buttonText={t('okay')} />
                </View>
            </View>
        );
    }
}
