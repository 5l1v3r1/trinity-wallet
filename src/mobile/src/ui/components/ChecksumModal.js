import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import { Styling } from 'ui/theme/general';
import { withNamespaces } from 'react-i18next';
import { width, height } from 'libs/dimensions';
import SingleFooterButton from './SingleFooterButton';

const styles = StyleSheet.create({
    modalContainer: {
        alignItems: 'center',
        width,
        height,
        justifyContent: 'flex-end',
    },
    modalContent: {
        justifyContent: 'space-between',
        alignItems: 'center',
        width,
        height: height - Styling.topbarHeight,
    },
    textContainer: {
        width: width - width / 10,
        paddingBottom: height / 22,
    },
    modalText: {
        color: 'white',
        fontFamily: 'SourceSansPro-Light',
        fontSize: Styling.fontSize3,
        textAlign: 'left',
        backgroundColor: 'transparent',
    },
    modalTextBold: {
        fontFamily: 'SourceSansPro-Bold',
        fontSize: Styling.fontSize3,
        textAlign: 'left',
        backgroundColor: 'transparent',
    },
});

export class ChecksumModal extends PureComponent {
    static propTypes = {
        /** @ignore */
        t: PropTypes.func.isRequired,
        /** Close active modal */
        closeModal: PropTypes.func.isRequired,
        /** @ignore */
        theme: PropTypes.object.isRequired,
    };

    render() {
        const { t, theme: { body } } = this.props;

        return (
            <View style={styles.modalContainer}>
                <View style={[styles.modalContent, { backgroundColor: body.bg }]}>
                    <View style={{ flex: 1 }} />
                    <View style={styles.textContainer}>
                        <Text style={[styles.modalTextBold, { color: body.color }, { paddingTop: height / 40 }]}>
                            {t('saveYourSeed:whatIsChecksum')}
                        </Text>
                        <Text style={[styles.modalText, { color: body.color }, { paddingTop: height / 60 }]}>
                            {t('saveYourSeed:checksumExplanation')}
                        </Text>
                    </View>
                    <View style={{ flex: 1 }} />
                    <SingleFooterButton onButtonPress={() => this.props.closeModal()} buttonText={t('okay')} />
                </View>
            </View>
        );
    }
}

export default withNamespaces(['logoutConfirmationModal', 'global'])(ChecksumModal);
