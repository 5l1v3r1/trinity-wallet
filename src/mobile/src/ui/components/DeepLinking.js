import get from 'lodash/get';
import React, { Component } from 'react';
import { Linking } from 'react-native';
import { withTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { initiateDeepLinkRequest, setDeepLinkContent, setSetting } from 'shared-modules/actions/wallet';
import { parseAddress, ADDRESS_LENGTH } from 'shared-modules/libs/iota/utils';
import { generateAlert } from 'shared-modules/actions/alerts';
import { getActiveTransaction } from 'shared-modules/selectors/exchanges/MoonPay';
import { fetchTransactionDetails } from 'shared-modules/actions/exchanges/MoonPay';
import { changeHomeScreenRoute } from 'shared-modules/actions/home';
import { MOONPAY_RETURN_URL } from 'shared-modules/exchanges/MoonPay';
import { isAndroid } from 'libs/device';
import navigator from 'libs/navigation';

export default () => (C) => {
    class WithDeepLinking extends Component {
        constructor() {
            super();
            this.setDeepUrl = this.setDeepUrl.bind(this);
        }

        componentDidMount() {
            Linking.addEventListener('url', this.setDeepUrl);
            if (isAndroid) {
                Linking.getInitialURL().then((url) => {
                    if (url) {
                        this.setDeepUrl(url);
                    }
                });
            }
        }

        componentWillUnmount() {
            Linking.removeEventListener('url', this.setDeepUrl);
        }

        /**
         * Prefills transactional information on deep link
         * @param {Object} data
         */
        setDeepUrl(data) {
            const { activeTransaction, t, generateAlert, deepLinking } = this.props;
            if (get(data, 'url').includes(MOONPAY_RETURN_URL)) {
                this.props.fetchTransactionDetails(get(activeTransaction, 'id'));
                return navigator.setStackRoot('reviewPurchase');
            }
            this.props.initiateDeepLinkRequest();
            if (!deepLinking) {
                this.navigateToSettings();
                return generateAlert('info', t('deepLink:deepLinkingInfoTitle'), t('deepLink:deepLinkingInfoMessage'));
            }
            this.props.changeHomeScreenRoute('send');
            const parsedData = parseAddress(data.url || data);
            if (parsedData) {
                this.props.setDeepLinkContent(
                    parsedData.amount.toString() || '0',
                    parsedData.address,
                    parsedData.message || null,
                );
            } else {
                generateAlert(
                    'error',
                    t('send:invalidAddress'),
                    t('send:invalidAddressExplanation1', { maxLength: ADDRESS_LENGTH }),
                );
            }
        }

        /**
         * Navigates to deep linking settings
         */
        navigateToSettings() {
            this.props.changeHomeScreenRoute('settings');
            this.props.setSetting('deepLinking');
        }

        render() {
            return <C {...this.props} />;
        }
    }

    WithDeepLinking.propTypes = {
        /** @ignore */
        t: PropTypes.func.isRequired,
        /** @ignore */
        generateAlert: PropTypes.func.isRequired,
        /** @ignore */
        setSetting: PropTypes.func.isRequired,
        /** @ignore */
        initiateDeepLinkRequest: PropTypes.func.isRequired,
        /** @ignore */
        setDeepLinkContent: PropTypes.func.isRequired,
        /** @ignore */
        changeHomeScreenRoute: PropTypes.func.isRequired,
        /** @ignore */
        fetchTransactionDetails: PropTypes.func.isRequired,
        /** @ignore */
        deepLinking: PropTypes.bool.isRequired,
        /** @ignore */
        activeTransaction: PropTypes.object,
    };

    const mapStateToProps = (state) => ({
        deepLinking: state.settings.deepLinking,
        activeTransaction: getActiveTransaction(state),
    });

    const mapDispatchToProps = {
        initiateDeepLinkRequest,
        setDeepLinkContent,
        generateAlert,
        changeHomeScreenRoute,
        setSetting,
        fetchTransactionDetails,
    };

    return withTranslation(['global'])(
        connect(
            mapStateToProps,
            mapDispatchToProps,
        )(WithDeepLinking),
    );
};
