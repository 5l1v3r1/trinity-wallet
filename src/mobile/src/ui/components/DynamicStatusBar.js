import React, { Component } from 'react';
import { StatusBar } from 'react-native';
import PropTypes from 'prop-types';
import tinycolor from 'tinycolor2';
import timer from 'react-native-timer';
import { Navigation } from 'react-native-navigation';
import { isAndroid, isIPhoneX } from 'libs/device';
import { rgbToHex } from 'shared-modules/libs/utils';

class DynamicStatusBar extends Component {
    static propTypes = {
        /** Status bar background color */
        backgroundColor: PropTypes.string,
        /** Determines whether modal is open */
        isModalActive: PropTypes.bool,
    };

    componentWillMount() {
        Navigation.events().registerComponentDidAppearListener(() => {
            this.resetStatusBarColor();
        });
    }

    componentDidMount() {
        this.resetStatusBarColor();
    }

    componentWillReceiveProps(newProps) {
        const { isModalActive } = this.props;
        if (!isAndroid) {
            return;
        }
        if (!isModalActive && newProps.isModalActive) {
            timer.setTimeout('timeout', () => this.resetStatusBarColor(), 50);
            if (isAndroid) {
                this.resetStatusBarColor();
            }
        }
        if (isModalActive && !newProps.isModalActive) {
            timer.setTimeout('timeout', () => this.resetStatusBarColor(), 450);
            if (isAndroid) {
                this.resetStatusBarColor();
            }
        }
    }

    getStatusBarStyle() {
        const { backgroundColor } = this.props;
        if (isIPhoneX) {
            return 'light-content';
        }
        return tinycolor(backgroundColor).isDark() ? 'light-content' : 'dark-content';
    }

    resetStatusBarColor() {
        const { backgroundColor } = this.props;
        if (backgroundColor) {
            StatusBar.setBackgroundColor(rgbToHex(backgroundColor), false);
        }
    }

    render() {
        const { backgroundColor } = this.props;
        const statusBarStyle = this.getStatusBarStyle();

        return <StatusBar barStyle={statusBarStyle} backgroundColor={backgroundColor} translucent animated={false} />;
    }
}
export default DynamicStatusBar;
