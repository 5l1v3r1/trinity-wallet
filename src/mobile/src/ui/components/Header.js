import React, { PureComponent } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Styling } from 'ui/theme/general';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
    },
    header: {
        fontFamily: 'SourceSansPro-Light',
        fontSize: Styling.fontSize6,
        textAlign: 'center',
        width: Styling.contentWidth,
    },
});

export default class Header extends PureComponent {
    static propTypes = {
        /* Text color for heading */
        textColor: PropTypes.string.isRequired,
        /* Heading text content */
        children: PropTypes.string.isRequired,
    };

    render() {
        const { textColor, children } = this.props;

        return (
            <View style={styles.container}>
                <Text style={[styles.header, { color: textColor }]}>{children}</Text>
            </View>
        );
    }
}
