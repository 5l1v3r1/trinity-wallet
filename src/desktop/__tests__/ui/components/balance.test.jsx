import React from 'react';
import { shallow } from 'enzyme';

import { Balance } from 'ui/components/Balance';

const props = {
    summary: false,
    index: 0,
    switchAccount: jest.fn(),
    seedIndex: 0,
    accounts: {
        accountInfo: {
            foo: {
                balance: 500,
            },
            bar: {
                balance: 100,
            },
        },
    },
    accountNames: ['foo', 'bar'],
    settings: {
        currency: 'USD',
        conversionRate: 1,
    },
    marketData: {
        usdPrice: 1,
    },
    t: (str) => str,
};

describe('Balance component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Balance {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Balance account name', () => {
        const wrapper = shallow(<Balance {...props} />);
        expect(wrapper.find('h3').text()).toEqual('foo');
    });
});
