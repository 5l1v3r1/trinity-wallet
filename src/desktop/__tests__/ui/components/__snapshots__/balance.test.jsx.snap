// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance component Render the component 1`] = `
<div
  className="balance"
>
  <h3>
    foo
  </h3>
  <h1
    onClick={[Function]}
  >
    500
    <small>
      i
    </small>
  </h1>
  <h2>
    $ 0.00
  </h2>
</div>
`;
